addons/sourcemod/scripting/vsh/bosses/boss_samyro.sp

```diff
@@ -0,0 +1,219 @@
+		if (this.bSuperRage)
+		{
+			CRageAddCond rageCond = this.CallFunction("FindAbility", "CRageAddCond");
+			if (rageCond != INVALID_ABILITY)
+			{
+				TF2_AddCondition(this.iClient, TFCond_Ubercharged, rageCond.flRageCondDuration * rageCond.flRageCondSuperRageMultiplier);
+			}
+		}
+	}
+	
+	public void GetSound(char[] sSound, int length, SaxtonHaleSound iSoundType)
+	{
+		switch (iSoundType)
+		{
+			case VSHSound_RoundStart: strcopy(sSound, length, g_strSamyroRoundStart[GetRandomInt(0, sizeof(g_strSamyroRoundStart) - 1)]);
+			case VSHSound_Win: strcopy(sSound, length, g_strSamyroRoundWin[GetRandomInt(0, sizeof(g_strSamyroRoundWin) - 1)]);
+			case VSHSound_Lose: strcopy(sSound, length, g_strSamyroRoundLose[GetRandomInt(0, sizeof(g_strSamyroRoundLose) - 1)]);
+			case VSHSound_Rage: strcopy(sSound, length, g_strSamyroRage[GetRandomInt(0, sizeof(g_strSamyroRage) - 1)]);
+			case VSHSound_Lastman: strcopy(sSound, length, g_strSamyroLastMan[GetRandomInt(0, sizeof(g_strSamyroLastMan) - 1)]);
+		}
+	}
+	
+	public void GetSoundAbility(char[] sSound, int length, const char[] sType)
+	{
+		if (strcmp(sType, "CAddCond") == 0)
+			strcopy(sSound, length, g_strSamyroAbility[GetRandomInt(0, sizeof(g_strSamyroAbility) - 1)]);
+	}
+	
+	public void GetSoundKill(char[] sSound, int length, TFClassType nClass)
+	{
+		strcopy(sSound, length, g_strSamyroKill[GetRandomInt(0, sizeof(g_strSamyroKill) - 1)]);
+	}
+	
+	public void GetMusicInfo(char[] sSound, int length, float &time)
+	{
+		strcopy(sSound, length, SAMYRO_MUSIC);
+		time = 195.0;
+	}
+	
+	public void Precache()
+	{
+		g_iSamyroModelHat = PrecacheModel("models/player/items/pyro/fwk_pyro_flamenco.mdl");
+		g_iSamyroModelMask = PrecacheModel("models/workshop/player/items/pyro/pyro_halloween_gasmask/pyro_halloween_gasmask.mdl");
+		g_iSamyroModelKatana = PrecacheModel("models/workshop_partner/weapons/c_models/c_shogun_katana/c_shogun_katana.mdl");
+		g_iSamyroModelHands = PrecacheModel("models/weapons/c_models/c_pyro_arms.mdl");
+		
+		PrepareSound(SAMYRO_MUSIC);
+		
+		for (int i = 0; i < sizeof(g_strSamyroRoundStart); i++) PrecacheSound(g_strSamyroRoundStart[i]);
+		for (int i = 0; i < sizeof(g_strSamyroRoundWin); i++) PrecacheSound(g_strSamyroRoundWin[i]);
+		for (int i = 0; i < sizeof(g_strSamyroRoundLose); i++) PrecacheSound(g_strSamyroRoundLose[i]);
+		for (int i = 0; i < sizeof(g_strSamyroRage); i++) PrecacheSound(g_strSamyroRage[i]);
+		for (int i = 0; i < sizeof(g_strSamyroKill); i++) PrecacheSound(g_strSamyroKill[i]);
+		for (int i = 0; i < sizeof(g_strSamyroLastMan); i++) PrecacheSound(g_strSamyroLastMan[i]);
+		for (int i = 0; i < sizeof(g_strSamyroAbility); i++) PrecacheSound(g_strSamyroAbility[i]);
+	}
+	
+	public bool IsBossHidden()
+	{
+		return true;
+	}
+};
+
+public Action Command_DropItem(int iClient, const char[] sCommand, int iArgs)
+{
+	//Prevent boss from dropping powerups
+	if (SaxtonHaleBase(iClient).bValid)
+		return Plugin_Handled;
+	
+	return Plugin_Continue;
+}
```

Comment:
May well then move this to `console.sp` if we gonna block `dropitem` to all bosses